code/modules/shuttle/misc/shuttle_remote.dm

```diff
@@ -0,0 +1,126 @@
+	///the port where the shuttle returns to
+	var/shuttle_home_id = "whiteship_home"
+	///var which will hold the nav computer
+	var/obj/machinery/computer/shuttle/our_computer
+	///var which will hold the mobile port
+	var/obj/docking_port/mobile/our_port
+
+/obj/item/shuttle_remote/examine(mob/user)
+	. = ..()
+	if(may_change_docks && our_computer)
+		. += span_notice("You can change where the [get_area_name(SSshuttle.getShuttle(our_computer.shuttleId))] docks using [EXAMINE_HINT("alt-right-click")].")
+
+/obj/item/shuttle_remote/Initialize(mapload)
+	. = ..()
+	if(!mapload)
+		return
+	var/obj/machinery/computer/shuttle/computer = locate(/obj/machinery/computer/shuttle) in loc
+	if(!computer)
+		log_mapping("[src] failed to find a navigation computer at [AREACOORD(src)]")
+		return
+	our_computer = computer
+	our_computer.remote_control = WEAKREF(src)
+
+/obj/item/shuttle_remote/interact_with_atom(atom/interacting_with, mob/living/user, list/modifiers)
+	if(!istype(interacting_with, /obj/machinery/computer/shuttle))
+		return NONE
+	if(our_computer || our_port)
+		balloon_alert(user, "already linked!")
+		return ITEM_INTERACT_BLOCKING
+	var/obj/machinery/computer/shuttle/new_computer = interacting_with
+	if(new_computer.remote_control || !new_computer.may_be_remote_controlled)
+		balloon_alert(user, "occupied signal!")
+		return ITEM_INTERACT_BLOCKING
+	new_computer.remote_control = WEAKREF(src)
+	our_computer = new_computer
+	our_port = SSshuttle.getShuttle(our_computer.shuttleId)
+	return ITEM_INTERACT_SUCCESS
+
+/obj/item/shuttle_remote/attack_self(mob/user)
+	if(!our_port)
+		our_port = SSshuttle.getShuttle(our_computer.shuttleId) //incase we were maploaded
+	if(!can_use(user))
+		return
+
+	var/obj/docking_port/home = SSshuttle.getDock(shuttle_home_id)
+	var/obj/docking_port/away = SSshuttle.getDock(shuttle_away_id)
+	var/obj/docking_port/dock = our_port.get_docked()
+
+	var/send_off_text = "Are you sure you want to send off [get_area_name(SSshuttle.getShuttle(our_computer.shuttleId))] to [away.name]?"
+	var/list/send_off_options = list("Yes", "No")
+	var/destination = null
+
+	if(home == dock || ("[our_computer.shuttleId]_custom" == dock.shuttle_id))
+		switch(tgui_alert(user, send_off_text, "Send Off Shuttle?", send_off_options))
+			if("Yes")
+				destination = away.shuttle_id
+	else if(away == dock)
+		send_off_text = "Are you sure you want to call [get_area_name(SSshuttle.getShuttle(our_computer.shuttleId))] to [home.name]?"
+		for(var/list/possible_destinations in our_computer.get_valid_destinations())
+			if(LAZYACCESS(possible_destinations, "id") == "[our_computer.shuttleId]_custom")
+				send_off_text += "\n\nCustom location loaded, try to dock?"
+				send_off_options += "Send to custom"
+				break
+		switch(tgui_alert(user, send_off_text, "Call Shuttle?", send_off_options))
+			if("Yes")
+				destination = home.shuttle_id
+			if("Send to custom")
+				destination = "[our_computer.shuttleId]_custom"
+
+	if(!destination)
+		return
```

Comment:
You need to check for input stalls. <https://github.com/tgstation/tgstation/blob/master/.github/guides/STANDARDS.md#develop-secure-code>

Otherwise, people can pick up the remote, open the tgui, and just hold the alert on their screen to telekinetically call the shuttle even if they're dead.